apps/mobile-wallet/src/features/modals/AppModals.tsx

```diff
@@ -16,32 +16,91 @@ You should have received a copy of the GNU Lesser General Public License
+import { getElementName, isModalWrapped } from '~/features/modals/modalUtils'
+import NftGridModal from '~/features/nftsDisplay/NftGridModal'
+import NftModal from '~/features/nftsDisplay/NftModal'
+import CurrencySelectModal from '~/features/settings/CurrencySelectModal'
+import MnemonicModal from '~/features/settings/MnemonicModal'
+import WalletDeleteModal from '~/features/settings/WalletDeleteModal'
+import TransactionModal from '~/features/transactionsDisplay/TransactionModal'
 import { useAppSelector } from '~/hooks/redux'
+import SwitchNetworkModal from '~/screens/SwitchNetworkModal'
 
 const AppModals = () => {
   const openedModals = useAppSelector(selectAllModals)
 
-  if (openedModals.length === 0) return null
-
   return (
     <ModalsContainer pointerEvents={openedModals.length > 0 ? 'auto' : 'none'}>
       {openedModals.map((modal) => {
-        switch (modal.params.name) {
+        const { id, params } = modal
+
+        switch (params.name) {
           case 'BuyModal':
-            return <BuyModal id={modal.id} key={modal.id} />
+            return <BuyModal key={id} id={id} />
+          case 'FundPasswordReminderModal':
+            return <FundPasswordReminderModal key={id} id={id} />
+          case 'BackupReminderModal':
+            return <BackupReminderModal key={id} id={id} {...params.props} />
+          case 'SwitchNetworkModal':
+            return <SwitchNetworkModal key={id} id={id} {...params.props} />
+          case 'TransactionModal':
+            return <TransactionModal key={id} id={id} {...params.props} />
+          case 'NftModal':
+            return <NftModal key={id} id={id} {...params.props} />
+          case 'NftGridModal':
+            return <NftGridModal key={id} id={id} {...params.props} />
+          case 'WalletDeleteModal':
+            return <WalletDeleteModal key={id} id={id} {...params.props} />
+          case 'BiometricsWarningModal':
+            return <BiometricsWarningModal key={id} id={id} {...params.props} />
+          case 'MnemonicModal':
+            return <MnemonicModal key={id} id={id} {...params.props} />
+          case 'AutoLockOptionsModal':
+            return <AutoLockOptionsModal key={id} id={id} />
+          case 'CurrencySelectModal':
+            return <CurrencySelectModal key={id} id={id} />
+          case 'LanguageSelectModal':
+            return <LanguageSelectModal key={id} id={id} />
+          default:
+            return null
         }
       })}
     </ModalsContainer>
   )
 }
 
+interface ModalsContainerProps {
+  children: ReactNode
+}
+
+const ModalsContainer = ({ children, ...props }: ModalsContainerProps & ViewProps) => {
+  useEffect(() => {
+    Children.forEach(children, (child) => {
+      if (isValidElement(child) && !isModalWrapped(child)) {
+        console.warn(
+          `Warning: ${getElementName(child)} is not wrapped! Please wrap it with the withModalWrapper function.`
```

Comment:
nit: is there a strong argument for the naming `withModalWrapper`? I find the name a bit off, I think I'd prefer something simpler like `modal` or `memoModal` or `modalMemo` or `memoizedModal`. WDYT?

Reply:
`withSomething` is a convention I found for HoCs. We could go for just `modal`.

Reply:
I see. I did some research and also found the `with` prefix. That's fine then. What about dropping the `Wrapper` parth tho? 😄

Reply:
`withModal` is weird to me 🙃 
I'd rather go with `modal` only, like `memo`.

Reply:
Changed my mind and went with `withModal`. Having a file called `modal.tsx` was just weird.
`with` makes it clear that it's a HOC.

Reply:
There is a lint error here

Reply:
No more lint error in part 3 PR.